tokio/src/net/tcp/split.rs

```diff
@@ -141,9 +141,9 @@ impl ReadHalf<'_> {
 
     /// Waits for any of the requested ready states.
     ///
-    /// This function is usually paired with `try_read()` or `try_write()`. It
-    /// can be used to concurrently read / write to the same socket on a single
-    /// task without splitting the socket.
+    /// This function is usually paired with [`try_read()`]. It can be used instead
+    /// of [`readable()`] to check the returned ready set for [`Ready::READABLE`]
+    /// and [`Ready::READ_CLOSED`] events.
```

Comment:
The `ready` methods are also sometimes paired with both `try_read` and `try_write` when you want to listen for both types of readiness. It's not just used as a more powerful version of `readable`.

Reply:
I agree that it can be paired like that, there is even an [example](https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html#examples-7) for `TcpStream` which reflects what you say.

My point is that in the case of `ReadHalf` we have no `try_write` to pair with, it only exists on the matching `WriteHalf`. To me, it seems weird to suggest pairing `read_half.ready()` with `write_half.try_write()`, or whatever the current docs are trying to tell. If a socket had to be split, then I would say it's likely that we cannot access both halves at the same time and use these methods like that.

This doc change is not a big deal, I just thought that it may be misleading. If it's intended to use split sockets like that, then I'm fine with leaving the docs as they are.

Reply:
Oh, I didn't notice this doc change was specific to split halves.